### ISSUE TITLE
Error messages passed to multiple fields in batch mode

### ISSUE THREAD

**Issue by jarro** [deadpixelsociety/passport#1]
I used the sample code and applied it to two fields.
When I validate in batch mode,  I end up with all three error messages on both fields.

Thanks for making this available,  it is just what I was looking for.

**Comment by deadpixelsociety** [comment]
Hi jarro! Would you be able to provide an example reproduction? Thanks.

**Comment by jarro** [comment]
```

private val validator by lazy {
    passport {

        rules<String?>(view?.ti_username!!) {
            email()
            length(8, 32)
        }

        rules<String?>(view?.ti_middle!!) {
            numeric()
        }
    }
}

private fun validate(): Boolean{
    return validator.validate(view!!, ValidationMethod.IMMEDIATE)
}


```

With something like that,  the numeric rule is applied to ti_username.   Thought it was related to ValidationMethod.BATCH  but  happens with immediate as well.   Seems like every rule is applied to every field.

**Comment by deadpixelsociety** [comment]
Thanks, I'll take a look.

**Comment by deadpixelsociety** [comment]
jarro,
I've addressed this in the 2.0.1 release just now. What happened was that validators were being treated as singleton instances and being shared across views. I've changed this so that validators are now instanced from factory functions via Passport.validatorFactory(). 

I've updated the example activity to show this.

Thanks for the report!

**Comment by jarro** [comment]
Working great,  thanks for the speedy fix.
